staging/src/k8s.io/apiserver/pkg/endpoints/handlers/finisher/finisher.go

```diff
@@ -62,10 +67,27 @@ func (r *result) Return() (runtime.Object, error) {
 	}
 }
 
+// PostTimeoutLoggerFunc is a function that can be used to log the result returned
+// by a ResultFunc after the request had timed out.
+// timedOutAt is the time the request had been timed out.
+// r is the result returned by the child goroutine.
+type PostTimeoutLoggerFunc func(timedOutAt time.Time, r *result)
+
+const (
+	// how much time the post-timeout receiver goroutine will wait for the sender
+	// (child goroutine executing ResultFunc) to send a result after the request.
+	// had timed out.
+	postTimeoutLoggerWait = 30 * time.Minute
```

Comment:
Why is this 30 minutes?

Reply:
after the request had timed out, we wait up to 30 minutes for the sender to return a result instead of waiting indefinitely. `30m` seems to be a good starting point.

Reply:
It seems like a huge period of time.. What kind of request takes 30 minutes to process?

Reply:
@logicalhan  The apiserver will timeout a request after `60s`, the caller is going to receive a `504`.  This post-timeout receiver kicks in after the request had been timed out, it waits for `30m` to receive a result from the executing request handler.

> What kind of request takes 30 minutes to process?
No request should take 30m to process. The purpose of the post-timeout receiver is to track how long a request handler is active after the request had been timed out. But we don't want to wait forever, `30m` seems to be good starting point.

One example, due to underlying network conditions (stale socket) an http response read/write could block for up to 15 to 20 minutes (depending on kernel settings).

Reply:
`30m` seems high since the default timeout is `60s` anything beyond that is concerning. I would expect the context to propagate the cancellation signal and to clean up resources.

From what I have seen `FinishRequest` is used to wrap the storage calls. I would not expect these calls to last 30 minutes.